src/default.jl

```diff
@@ -199,9 +200,19 @@ function defaultalg(A, b, assump::OperatorAssumptions)
     elseif assump.condition === OperatorCondition.WellConditioned
         DefaultAlgorithmChoice.NormalCholeskyFactorization
     elseif assump.condition === OperatorCondition.IllConditioned
-        DefaultAlgorithmChoice.QRFactorization
+        if is_underdetermined(A)
+            # Underdetermined
+            DefaultAlgorithmChoice.QRFactorizationPivoted
```

Comment:
QRFactorization is already pivoted?

Reply:
oh I see. @YingboMa what do you think of just always defaulting to pivoted QR?

Reply:
Yeah, pivoted QR handles degenerate systems too.

Reply:
But it's quite a bit slower though.